chromium_src/python_modules/components/policy/tools/template_writers/brave_writer_configuration.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+# Copyright 2022 The Brave Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import unittest
+import writer_configuration
+
+def GetConfigurationForBuild(defines):
+    base = writer_configuration.GetConfigurationForBuild({'_chromium'})
+    return _merge_dicts(_BRAVE_VALUES, base)
+
+_BRAVE_VALUES = {
+    'build': 'brave',
+    'app_name': 'Brave',
+    'doc_url':
+        'https://support.brave.com/hc/en-us/articles/360039248271-Group-Policy',
+    'frame_name': 'Brave Frame',
+    'webview_name': 'Brave WebView',
+    'win_config': {
+        'win': {
+            'reg_mandatory_key_name':
+                'Software\\Policies\\BraveSoftware\\Brave',
+            'reg_recommended_key_name':
+                'Software\\Policies\\BraveSoftware\\Brave\\Recommended',
+            'mandatory_category_path': ['Brave:Cat_Brave', 'brave'],
+            'recommended_category_path': ['Brave:Cat_Brave',
+                                          'brave_recommended'],
+            'category_path_strings': {
+                'brave': 'Brave',
+                'brave_recommended':
+                'Brave - {doc_recommended}'
+            },
+            'namespace': 'BraveSoftware.Policies.Brave',
+        },
+    },
+    # The string 'Brave' is defined in brave.adml for ADMX, but ADM doesn't
+    # support external references, so we define this map here.
+    'adm_category_path_strings': {
+        'Brave:Cat_Brave': 'Brave'
+    },
+    'admx_prefix': 'brave',
+    'admx_using_namespaces': {
+        'Brave': 'BraveSoftware.Policies'  # prefix: namespace
+    },
+    'linux_policy_path': '/etc/brave/policies/',
+    'bundle_id': 'com.brave.ios.core',
+}
+
+def _merge_dicts(src, dst):
+    result = dict(dst)
+    for k, v in src.items():
+        result[k] = \
+            _merge_dicts(v, dst.get(k, {})) if isinstance(v, dict) else v
+    return result
+
+class MergeDictsTest(unittest.TestCase):
+    def test_merge(self):
+        self.assertEqual(
+            # We actually only care about 'brave' being in the result.
+            {'brave': 'Brave', "googlechrome": "Google Chrome"},
+            _merge_dicts({'brave': 'Brave'}, {"googlechrome": "Google Chrome"})
+        )
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
?

Reply:
Thank you for the review :) Are you asking why `unittest.main()` is here? Or are you asking why there's test code at all?

Reply:
> why `unittest.main()` is here?

I guess this. I missed `MergeDictsTest` and thought that this call was a leftover, but now I see that along with the production code here lies the test. Not sure if this the right approach.

Reply:
So what would you advise?

1. Remove just `unittest.main()`
2. Remove the test as well.
3. Move the test to another file; If yes - which one?
4. Do nothing. Keep as-is.

The test isn't great so I'd be OK with dropping it. However, I did find it useful. So my preference would be 4. (otherwise I wouldn't have written it this way).

Reply:
I think the better approach is to create a file `brave_writer_configuration_tests.py` in the same folder and put tests here. This file can import `brave_writer_configuration` and test the function you need.

Reply:
Ok, will do it this way. Thanks!

Reply:
It didn't work to put the tests into a separate file. The reason is that `bwc_tests.py` has to import `bwc.py`. But that latter file needs to have `@override_utils.override_function` at the top, redefining a Chromium function. That Chromium function is not (easily) importable when the unit test is run, leading to an error.

I've simply removed the test code now. It wasn't worth the effort.